src/reducers/reducer.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+const _defaultState = {
+	shoppingCart: [],
+	shirtCounter: 0,
+	shoesCounter: 0,
+	hatCounter: 0,
+	glassCounter: 0
+}
+
+const reducer = (state = _defaultState, action) => {
+	switch(action.type){
+		case "ADD_ITEM":
+			if 	(action.item === "shirt"){
```

Comment:
should have abstracted the Object.assign outside of the if conditional. Within each conditional you could have manipulated and returned what you wanted.